tests/cucumber-reportportal-formatter.spec.js

```diff
@@ -226,23 +647,170 @@ describe('cucumber-reportportal-formatter', () => {
+      const spyfinishTestItem = jest.spyOn(formatter.reportportal, 'finishTestItem');
+
+      formatter.onTestStepFinishedEvent({
+        ...testStepFinished,
+        testStepResult: { ...testStepFinished.testStepResult, status: STATUSES.PASSED },
+      });
+
+      expect(spyfinishTestItem).toBeCalledWith(
+        'testItemId',
+        expect.objectContaining({
+          status: STATUSES.PASSED,
+        }),
+      );
+      expect(formatter.storage.getStepTempId(testStepStarted.testStepId)).toBeUndefined();
+    });
+
+    it('finishTestItem should be called with unexpected status', () => {
+      const spyfinishTestItem = jest.spyOn(formatter.reportportal, 'finishTestItem');
+
+      formatter.onTestStepFinishedEvent({
+        ...testStepFinished,
+        testStepResult: { ...testStepFinished.testStepResult, status: 'unexpected-status' },
+      });
+
+      expect(spyfinishTestItem).toBeCalledWith(
+        'testItemId',
+        expect.objectContaining({
+          status: 'unexpected-status',
+        }),
+      );
+    });
+
+    it('finishTestItem should be called with NOT_ISSUE type if SKIPPED status', () => {
+      const spyfinishTestItem = jest.spyOn(formatter.reportportal, 'finishTestItem');
+
+      formatter.config.skippedIssue = false;
+      formatter.onTestStepFinishedEvent({
+        ...testStepFinished,
+        testStepResult: { ...testStepFinished.testStepResult, status: STATUSES.SKIPPED },
+      });
+
+      expect(spyfinishTestItem).toBeCalledWith(
+        'testItemId',
+        expect.objectContaining({ status: STATUSES.SKIPPED, issue: { issueType: 'NOT_ISSUE' } }),
+      );
+    });
+
+    it('should set failed status for test case in storage if isScenarioBasedStatistics === true', () => {
+      const originIsScenarioBasedStatistics = formatter.isScenarioBasedStatistics;
+      formatter.isScenarioBasedStatistics = true;
 
       formatter.onTestStepFinishedEvent(testStepFinished);
+      formatter.isScenarioBasedStatistics = originIsScenarioBasedStatistics;
 
-      expect(spyFinishTestItem).toBeCalledWith('testItemId', {
-        endTime: mockedDate,
-        status: STATUSES.FAILED,
-        description: '```error\nerror message\n```',
+      expect(formatter.storage.getTestCase(testCaseId).status).toEqual(STATUSES.FAILED);
+    });
+
+    it('finishTestItem should be called with NOT_ISSUE type', () => {
+      const spyfinishTestItem = jest.spyOn(formatter.reportportal, 'finishTestItem');
+
+      formatter.onTestStepFinishedEvent({
+        ...testStepFinished,
+        testStepResult: { ...testStepFinished.testStepResult, status: STATUSES.PASSED },
       });
+
+      expect(spyfinishTestItem).toBeCalledWith(
+        'testItemId',
+        expect.objectContaining({
+          status: STATUSES.PASSED,
+        }),
+      );
       expect(formatter.storage.getStepTempId(testStepStarted.testStepId)).toBeUndefined();
     });
+
+    it('sendLog should be called with pending message', () => {
```

Comment:
I would suggest to add a 'in case of corresponding status received' at the end of the test name.
Could be added to 2 other similar tests.